### DISCUSSION TITLE
Less transfer of tensors between CPU-GPU.

### DISCUSSION THREAD

**Discussion by eelgaev** [neonbjb/tortoise-tts#251] [Ideas]
First of all, this is currently the best TTS project on github, and James has done amazing work here.

I played a little with the code and noticed there is many CPU operations done in pytorch, we finish inferencing on the autoregression model and we move it back to the CPU before running the other models,  (I believe) I was able to move all of the operations to the GPU only, and it is much faster now.

The fork is here:
https://github.com/Okoyl/tortoise-tts-nocpu

I'd like some feedback, hopefully it is good enough for a PR.

**Comment by neonbjb** [comment]
This will greatly speed things up, but requires quite a bit of gpu memory.
Thanks for posting this!

**Comment by I-Have-No-Idea-What-IAmDoing** [comment]
How does it compare to [metavoicexyz solution](https://github.com/metavoicexyz/tortoise-tts)?

**Comment by vdruts** [comment]
Does this require a whole new rebuild or can we just replace some files.

**Reply by lavnishhh** [reply]
How do I clear the previous build?

**Comment by FurkanGozukara** [comment]
> First of all, this is currently the best TTS project on github, and James has done amazing work here.
> 
> I played a little with the code and noticed there is many CPU operations done in pytorch, we finish inferencing on the autoregression model and we move it back to the CPU before running the other models, (I believe) I was able to move all of the operations to the GPU only, and it is much faster now.
> 
> The fork is here: https://github.com/Okoyl/tortoise-tts-nocpu
> 
> I'd like some feedback, hopefully it is good enough for a PR.

How much vram it uses? 12 vram enough?

**Reply by neonbjb** [reply]
I tested that it works on 12gb gpus.

**Reply by lavnishhh** [reply]
> I tested that it works on 12gb gpus.

I have 4GB VRAM. I assume there won't be any difference? :)
